contract.go

```diff
@@ -34,18 +36,46 @@ func (c *Contract) Txn(method string, opts TxnOpts, args ...interface{}) (string
 		return "", fmt.Errorf("failed to create contract transaction: %w", err)
 	}
 
+	gasPrice, err := c.Client.GasPrice()
+	if err != nil {
+		return "", fmt.Errorf("failed to get gas price: %w", err)
+	}
+	blockNumber, err := c.Client.BlockNumber()
+	if err != nil {
+		return "", fmt.Errorf("failed to get block number: %w", err)
+	}
+	block, err := c.Client.GetBlockByNumber(ethgo.BlockNumber(blockNumber), false)
+	if err != nil {
+		return "", fmt.Errorf("failed to get block: %w", err)
+	}
+
 	txo := contract.TxnOpts{
 		Value:    big.NewInt(int64(opts.Value)),
-		GasPrice: opts.GasPrice,
-		GasLimit: opts.GasLimit,
+		GasPrice: gasPrice,
+		GasLimit: block.GasLimit,
 		Nonce:    opts.Nonce,
 	}
 	txn.WithOpts(&txo)
 
 	err = txn.Do()
 	if err != nil {
-		return "", fmt.Errorf("failed to send contract transaction: %w", err)
+		return "", fmt.Errorf("failed to send contract transaction: %w, Tx: %+v", err, txo)
 	}
 
 	return txn.Hash().String(), nil
 }
+
+func (c *Contract) GetAddress() string {
+	return c.SignerAddress
+}
+
+func (c *Contract) FillInput(abiString string, method string) []byte {
```

Comment:
this function should contain function parameters as I understand

Reply:
yes, it was not full
I completed it with full input calculation for given method and args values